src/core/tests/SO3UtilsUnitTest.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia
+ * Authors: Stefano Dafarra
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#include <iDynTree/Core/Utils.h>
+#include <iDynTree/Core/TestUtils.h>
+#include <iDynTree/Core/SO3Utils.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <Eigen/Dense>
+
+#include <cmath>
+
+void checkGeodesicDistance()
+{
+    iDynTree::Rotation r1 = iDynTree::getRandomRotation();
```

Comment:
https://github.com/robotology/idyntree/blob/d5bbceab9e8ae52943b31199f6e024d12ff52159/src/core/src/TestUtils.cpp#L101

Random rotations here choose a value in the range [-10, 10]. I am not sure if these will cause any unexpected singularities due to conversion from RPY angles to rotation matrices? I think it's better to check again if determinant of these random rotations are 1 and the Frobenius norm is not either Inf or NaN

Reply:
I don't think there is  any problem if you get a RPY value at a singularity point, as that will result in a normal Rotation. The problem of  singularity configurations is that there are infinite  values of RPY angles that give the same Rotation, but this does not mean that the resulting Rotation is ill-defined or problematic in any way.

Reply:
Ah, that's true, indeed.